src/server/ServerHandler.java

```diff
@@ -37,6 +48,108 @@ public ServerHandler(Socket socket, String username) {
         this.currentUsername = username;
     }
 
+    /**
+     * Main request handler that processes client requests and returns appropriate responses.
+     *
+     * @param request the request object received from the client
+     * @return a response object based on the request type
+     */
+    public Response handleRequest(Request request) {
+        if (request == null) {
+            return new Response(ResponseStatus.FAILURE, "Request cannot be null");
+        }
+
+        // Use switch-case to decide among the two request types
+        switch (request.getType()) {
+            case SEND_MOVE:
+                return handleSendMoveRequest(request);
+            case REQUEST_MOVE:
+                return handleRequestMove();
+            default:
+                // Return failed response if neither of the two types is sent
+                return new Response(ResponseStatus.FAILURE, "Unsupported request type: " + request.getType());
+        }
+    }
+
+    /**
+     * Handles SEND_MOVE requests by deserializing the move data from the request
+     * and delegating to the handleSendMove(move) function for processing.
+     *
+     * @param request the SEND_MOVE request containing move data in the data attribute
+     * @return a response indicating success or failure of move processing
+     */
+    private Response handleSendMoveRequest(Request request) {
+        try {
+            // Deserialize the data attribute to get the move
+            int move = Integer.parseInt(request.getData());
+
+            // Call the separate function with just the move parameter
+            return handleSendMove(move);
+
+        } catch (NumberFormatException e) {
```

Comment:
Add some logging here

```suggestion
        } catch (NumberFormatException e) {
            LOGGER.warn("Invalid move format: {}", request.getData());
```